### PR TITLE
CD-511 Demo content and Github Actions

### PR DESCRIPTION
**Pull request by left23**
Many diffs come from updating the existing test environment with the most current from default `.github/workflows/run-tests.yml` from [drupal_starterkit](https://github.com/UN-OCHA/drupal-starterkit/tree/develop/.github/workflows), adding files and other dependencies.

This PR also adds the `unocha_demo_content` module and removes a sample database we used for Travis.

## Additions for Performance or e2e tests
I've added the following optional steps to `run-tests.yml`
Custom Github labels need to be applied to the PR for the above test steps to run: `performance` or `e2e` depending.
### Install demo content
Enables `unocha_demo_content` module and imports nodes, menu items etc from json, set the frontpage
### Set up node
Github action to install node
### Install node dependencies
In sub theme, install packages including Jest for e2e
### Run e2e tests
Runs Jest e2e tests in sub theme using as `ci` at `http://127.0.0.1:8080`
### Audit URLs using Lighthouse
Runs multiple times on 2 URLS (`http://127.0.0.1:8080` and `http://127.0.0.1:8080/demo`) against a performance budget and provides temporary links to Lighthouse reports.

### CODE REVIEW

**Comment by attiks** [comment]
> cp: can't stat 'docker/memcache.services.yml': No such file or directory

https://github.com/UN-OCHA/common-design-site/actions/runs/7180976505/job/19554391610?pr=348#step:12:427

**Comment by left23** [comment]
> > cp: can't stat 'docker/memcache.services.yml': No such file or directory
> 
> https://github.com/UN-OCHA/common-design-site/actions/runs/7180976505/job/19554391610?pr=348#step:12:427

Thanks!

**Comment by attiks** [comment]
You'll need a path for phpunit, add something like `./html/modules/custom/unocha_demo_content` add the end of the command

**Comment by left23** [comment]
> ./

@attiks Is there an example in another repo you can point to? I am getting the defaults from the https://github.com/UN-OCHA/drupal-starterkit/blob/develop/.github/workflows/run-tests.yml
I get the impression that the failure https://github.com/UN-OCHA/common-design-site/actions/runs/7183087977/job/19561043468 means there are no tests, but that phpunit is present. If there are no tests in this repo, how do I deal with this?
Maybe take a queue from the starter_module https://github.com/UN-OCHA/drupal-starterkit/tree/develop/html/modules/custom/starter_module or omit the test in the run-tests.yml?

**Comment by left23** [comment]
`No tests executed!`

I'm quitting on a win. Run Test step is working.

**Comment by left23** [comment]
I've added reviewers for awareness, everyone optional.

File: composer.json [hunk 1]
```diff
@@ -87,15 +184,45 @@
...
+        "release:major": "conventional-changelog --major",
         "git-hooks": "git config core.hooksPath git-hooks || true",
-        "npm-prepare": "cd html/themes/contrib/common_design && npm install",
-        "npm-cleanup": "cd html/themes/contrib/common_design && rm -rf ./node_modules",
+        "npm-prepare": "cd html/themes/custom/common_design_subtheme && npm install",
+        "npm-cleanup": "cd html/themes/custom/common_design_subtheme && rm -rf ./node_modules",
         "sub-theme": "test -d html/themes/custom/common_design_subtheme || (cp -r html/themes/contrib/common_design/common_design_subtheme html/themes/custom/ && mv html/themes/custom/common_design_subtheme/common_design_subtheme.info.yml.example html/themes/custom/common_design_subtheme/common_design_subtheme.info.yml)",
-        "e2e-test": "cd html/themes/contrib/common_design && npm run ci -- -t '^(?!.*Subtheme).*$'",
+        "e2e-test": "cd html/themes/custom/common_design_subtheme && npm run ci -- -t '^(?!.*Subtheme).*$'",
         "e2e": [
```
[the remark this answers is not in the document]
Comment (cafuego) [reply composer.json:226]: Yes, kindly no npm anything in site repositories.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-# Common Design Demo | [![Build Status](https://travis-ci.com/UN-OCHA/common-design-site.svg?branch=master)](https://travis-ci.com/UN-OCHA/common-design-site)
```
Comment (cafuego) [inline README.md:1]: 🤣

File: .github/workflows/run-tests.yml [hunk 3]
```diff
@@ -167,59 +243,25 @@ jobs:
             *Pusher: @${{ github.actor }}, Action: `${{ github.event_name }}`, Workflow: `${{ github.workflow }}`*
           edit-mode: replace
 
-      - name: Slack Success Notification
```
Comment (cafuego) [inline .github/workflows/run-tests.yml:170]: Wait, no slack notifications anymore?

Comment (left23) [reply .github/workflows/run-tests.yml:170]: Thanks, I will re-add them.

**Review by lazysoundsystem — APPROVED** [review]

**Pull request merged**